### ISSUE TITLE
Add running ballot tracker to county audit wizard

### ISSUE THREAD

**Issue by ranweiler** [FreeAndFair/ColoradoRLA#125]
Of ballots selected for audit, shows next ballot, upcoming, and allows download of the full list of ballots. Fixed page position wrt scrolling.

**Comment by kiniry** [comment]
This is an important UI feature for `phase-3`.  I'm hoping we get it in, as I believe everything is done on the server side.

**Comment by nealmcb** [comment]
Here are some data sources and specs to work from.

The server provides this via the following GET request:
```
$ main.py -E '/cvr-to-audit-list?start=0&ballot_count=2'
<Response [200]> GET /cvr-to-audit-list?start=0&ballot_count=2
<Response [200]> GET None [
  {
    "audit_sequence_number": 6,
    "scanner_id": 3,
    "batch_id": 800,
    "record_id": 9,
    "imprinted_id": "3-800-9",
    "cvr_number": 7,
    "db_id": 1484,
    "ballot_type": "55",
    "storage_location": "Bin 1"
  },
  {
    "audit_sequence_number": 5,
    "scanner_id": 4,
    "batch_id": 1200,
    "record_id": 27,
    "imprinted_id": "4-1200-27",
    "cvr_number": 65,
    "db_id": 1542,
    "ballot_type": "40",
    "storage_location": "Bin 1"
  }
]
```

I'd propose:

Header text: "Upcoming ballot cards to be audited".

followed by a box with `audit_sequence_number, imprinted_id, ballot_type, storage_location` if that will fit. We don't know how big the storage location (from the manifest) will be unfortunately, though the others are probably reasonably small numbers.

Optional: allow them to sort by either the original presentation sequence (which should be by the numeric values of the imprinted id numbers), or audit_sequence_number.

For now it should probably include data on the current round of ballots, and allow scrolling if they don't fit. It would probably be helpful some day to support letting them choose the size of the set to display (and thus the round size).

**Comment by nealmcb** [comment]
It probably also makes sense to include information on whether the given cvr has been audited or not. That would make it easier for counties to keep track of their status.

**Comment by dmzimmerman** [comment]
If it’s been audited, you don’t see it in that list. Or at least, you shouldn’t.

**Comment by nealmcb** [comment]
That's true. For my last suggestion to make sense, we would also want to add some sort of "include audited ballots" option, which would make much more sense for a list to be downloaded in csv format than for the running ballot tracker in the GUI.

**Comment by ranweiler** [comment]
I think we should add a column to the ballot list page that indicates (via check mark) if a given ballot has been audited or not.
 
In the ballot card verification screen, we explicitly refer to the table on the previous page, which (at the time of writing) _shrinks_ as ballots are audited. This is a bug: it should stay fixed, and show all ballots in the round. Why? Well, on the verification page, we are spec'd to refer to the current ballot as "ballot #K of N ballots that you must audit in this round", where `K` is a 1-based index into the list of all the round's ballots. Additionally, we want to present ballots in location order (i.e. the order of that table). If the list on the previous page shrinks, then either (1) the order number doesn't match the preceding page, but correctly refers to an invisible list of all the round's ballots, or (2) the order number _does_ match the preceding page (whose table is shrinking), and thus the order number is always "1".

Instead, we should _always_ list all the round's `N` ballots on the ballot list page. This makes the "ballot #K" part of the spec sensible. The checkmark "Audited" column will make it easy to visually scan where in the list the current ballot under audit is, as well as status of earlier ballots. It also addresses the situation @sfsinger19103 just commented with: they will always have the original list, in location order, so they can just re-print it.

**Comment by sfsinger19103** [comment]
I agree completely.

**Comment by ranweiler** [comment]
See also issues #630 (the "current ballot number" problem), and #660 (which stipulates the desired "current ballot" behavior).

**Comment by nealmcb** [comment]
In practice, for many audits I've been to, they create a tracking form for each ballot selected, and that form is used to help locate the ballot and stays with the ballot. This is how they address the story from @sfsinger19103 .

I've added that future possible feature as #671 Print a tracking form for each ballot card selected for audit

**Comment by sfsinger19103** [comment]
Just sent email to CDOS, asking about static vs shrinking list of ballots, and saying we'll do the static version unless we hear otherwise from them.

> Dear CDOS team,
> 
> On Screen 3 (per your Audit Board Reporting Utility Mockup, attached), there is a list of ballot cards. 
> 
> As the Audit Board progresses through the review of ballots, should that list show:
> all the cards in the round, including cards that have been reviewed
> only the cards in the round that have yet to be reviewed
> something else?
> In the first option, the list is the same whether you consult it at the beginning, middle, or end of the round.
> 
> In the second option, the list gets shorter as the round progresses.
> 
> If we don't hear from you, we will implement the first option (list stays the same throughout the round).
>

**Comment by ranweiler** [comment]
Note: this _may_ be closed in #672, pending CDOS feedback.

**Comment by ranweiler** [comment]
This is closed in #672, since CDOS chose the first option.
